net/pfSense-pkg-freeradius2/files/usr/local/pkg/freeradius.inc

```diff
@@ -4409,4 +4415,834 @@ EOD;
+	}
+
+	// Maximum Requests Per Server
+	if ($post['varsettingsmaxrequestsperserver'] != '' && !is_numericint($post['varsettingsmaxrequestsperserver'])) {
+		$input_errors[] = "The 'Maximum Requests Per Server' field must contain an integer value.";
+	}
+
+	// OTP Lifetime
+	if ($post['varsettingsmotpenable'] == 'on') {
+		if ($post['varsettingsmotptimespan'] === 0) {
+			$input_errors[] = "The 'OTP Lifetime' field must not be 0.";
+		} elseif (empty($post['varsettingsmotptimespan'])) {
+			$input_errors[] = "The 'OTP Lifetime' field must not be empty.";
+		} elseif (!is_numericint($post['varsettingsmotptimespan'])) {
+			$input_errors[] = "The 'OTP Lifetime' field must contain an integer value.";
+		} elseif ($post['varsettingsmotptimespan'] > 12) {
+			$input_errors[] = "The 'OTP Lifetime' field should contain only sane secure values. Values higher than 12 (~120 seconds) are not allowed.";
+		}
+	}
+
+	// Number of Invalid Password Attempts
+	if ($post['varsettingsmotpenable'] == 'on') {
+		if ($post['varsettingsmotppasswordattempts'] === 0) {
+			$input_errors[] = "The 'Invalid Password Attempts' field must not be 0.";
+		} elseif (empty($post['varsettingsmotppasswordattempts'])) {
+			$input_errors[] = "The 'Invalid Password Attempts' field must not be empty.";
+		} elseif (!is_numericint($post['varsettingsmotppasswordattempts'])) {
+			$input_errors[] = "The 'Invalid Password Attempts' field must contain an integer value.";
+		} elseif ($post['varsettingsmotppasswordattempts'] > 20) {
+			$input_errors[] = "The 'Invalid Password Attempts' field should contain only sane secure values. Values higher than 20 are not allowed.";
+		}
+	}
+
+	// Token Password Length
+	if ($post['varsettingsmotptokenlength'] != '') {
+		if (!preg_match('/^[0-9]{1,2}(?:[-][0-9]{1,2}){1}$/i', $post['varsettingsmotptokenlength'])) {
+			$input_errors[] = "The 'Token Password Length' field must contain a valid integer range between 1 and 16, separated with a '-' character; e.g.: 1-6.";
+		} elseif (!is_intrange($post['varsettingsmotptokenlength'], 1, 16)) {
+			$input_errors[] = "The 'Token Password Length' field must contain a valid integer range between 1 and 16, separated with a '-' character; e.g.: 1-6.";
+		}
+	}
+
+	/*
+	 * TODO:
+	 * Logging Configuration - Additional Information for Bad Attempts, Additional Information for Good Attempts
+	 * Miscellaneous Configuration - Default Profile, Profile Attribute, Access Attribute
+	 * Group Membership Options - Groupname Attribute, Group Membership Filter, Group Membership Attribute
+	 */
+
+}
+
+/* EAP settings input validation */
+function freeradius_validate_eap($post, &$input_errors) {
+
+	// Disable Weak EAP Types sanity check
+	if ($post['vareapconfdisableweakeaptypes'] == 'on') {
+		if ($post['vareapconfdefaulteaptype'] == 'md5' || $post['vareapconfdefaulteaptype'] == 'gtc' || $post['vareapconfdefaulteaptype'] == 'leap') {
```

Comment:
For three tests this could go either way but it would be easier/cleaner to use in_array here, e.g. `in_array($post['vareapconfdefaulteaptype'], array('md5', 'gtc', 'leap'))`.